### PR TITLE
[WIP][Drup-354] Display plans.

### PR DESCRIPTION
**Pull request by Jaesin**
This work also contains the rate plan entity.

### CODE REVIEW

File: apigee_m10n.links.menu.yml [hunk 1]
```diff
@@ -14,3 +14,10 @@ apigee_m10n.packages:
   route_name: apigee_monetization.my_packages
   menu_name: account
   weight: -20
+apigee_m10n.rate_plan:
+  title: 'Rate Plans'
```
Comment (cnovak) [inline apigee_m10n.links.menu.yml]:
```suggestion
  title: 'Rate plans'
```

**Review by cnovak — COMMENTED** [review]
When going to user/1/monetization/packages URL, I am getting:
```
Error: Call to a member function add() on null in Drupal\apigee_m10n\Entity\RatePlan->set() (line 121 of modules/contrib/apigee_m10n/src/Entity/FieldableMonetizationEntityBaseTrait.php).

Drupal\apigee_m10n\Entity\RatePlan->set('keepOriginalStartDate', 1) (Line: 189)
Drupal\apigee_m10n\Entity\Storage\FieldableMonetizationEntityStorageBase->initFieldValues(Object, Array) (Line: 151)
Drupal\apigee_m10n\Entity\Storage\FieldableMonetizationEntityStorageBase->doCreate(Array) (Line: 221)
...
```

File: src/Entity/FieldableMonetizationEntityBaseTrait.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ * @see \Drupal\apigee_edge\Entity\FieldableEdgeEntityInterface
+ */
+trait FieldableMonetizationEntityBaseTrait {
+
+  use FieldableEdgeEntityBaseTrait {
+    set as edgeSet;
+  }
+
+  /**
+   * This is mostly copied from `FieldableEdgeEntityBaseTrait` but we are adding
```
Comment (mxr576) [inline src/Entity/FieldableMonetizationEntityBaseTrait.php:47]: Could you add this feature without code duplication? :skull_and_crossbones:
For example by introducing a base class for mint entities that extends the inherited set() method from this trait?

File: src/Entity/Storage/MonetizationEntityStorageBase.php [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityStorageBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+abstract class MonetizationEntityStorageBase extends EntityStorageBase {
```
Comment (mxr576) [inline src/Entity/Storage/MonetizationEntityStorageBase.php:35]: This is also a copy-paste clone :skull_and_crossbones:  of https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/src/Entity/Storage/EdgeEntityStorageBase.php. Why did you extended the EdgeEntityStorageBase class instead?

File: src/Entity/FieldableMonetizationEntityBaseTrait.php (hunk 2 above)
Comment (Jaesin) [reply src/Entity/FieldableMonetizationEntityBaseTrait.php:47]: There have been some recent changes to `apigee_edge` that breaks what we are doing here. We will see what we can to about this when we pull in those changes and make the necessary fixes.

**Comment by cnovak** [comment]
@mxr576 I created issues to fix the MonetizationEntityStorageBase.php and FieldableMonetizationEntityBaseTrait.php as you commented above.

https://apigeesc.atlassian.net/browse/DRUP-424
https://apigeesc.atlassian.net/browse/DRUP-425

**Comment by zakiya** [comment]
I signed the CLA.

**Pull request closed without merging**